src/Domain/Creators/BinaryCompactObject.cpp

```diff
@@ -207,18 +239,30 @@ BinaryCompactObject::BinaryCompactObject(
       }
     }
   };
-  add_object_region("ObjectA", "Shell");  // 6 blocks
-  add_object_region("ObjectA", "Cube");   // 6 blocks
-  add_object_region("ObjectB", "Shell");  // 6 blocks
-  add_object_region("ObjectB", "Cube");   // 6 blocks
-  add_outer_region("Envelope");           // 10 blocks
-  add_outer_region("OuterShell");         // 10 blocks
-  if (not object_A_.is_excised()) {
+  if (not use_single_block_a) {
```

Comment:
Swap `if/else` cases to avoid the negation (negations are surprisingly hard to read / understand)

Reply:
Do this throughout all files as well. Also when you do need to negate something, prefer `not` over `!`